nft-example/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::{H256, U256};
+use scale_info::TypeInfo;
+
+pub mod payloads;
+pub use payloads::{ApproveForAllInput, ApproveInput, InitConfig, TransferInput};
+
+pub mod state;
+pub use state::{State, StateReply};
+
+use non_fungible_token::base::NonFungibleTokenBase;
+use non_fungible_token::{Approve, ApproveForAll, NonFungibleToken, Transfer};
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug, Decode, TypeInfo)]
+pub enum Action {
+    Mint,
+    Burn(U256),
+    Transfer(TransferInput),
+    Approve(ApproveInput),
+    ApproveForAll(ApproveForAllInput),
+    OwnerOf(U256),
+    BalanceOf(H256),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    Transfer(Transfer),
+    Approval(Approve),
+    ApprovalForAll(ApproveForAll),
+    OwnerOf(H256),
+    BalanceOf(U256),
+}
+
+#[derive(Debug)]
+pub struct NFT {
+    pub token: NonFungibleToken,
+    pub token_id: U256,
+    pub owner: ActorId,
+}
+
+static mut CONTRACT: NFT = NFT {
+    token: NonFungibleToken {
+        name: String::new(),
+        symbol: String::new(),
+        base_uri: String::new(),
+        owner_by_id: BTreeMap::new(),
+        token_metadata_by_id: BTreeMap::new(),
+        token_approvals: BTreeMap::new(),
+        balances: BTreeMap::new(),
+        operator_approval: BTreeMap::new(),
+    },
+    token_id: U256::zero(),
+    owner: ActorId::new(H256::zero().to_fixed_bytes()),
+};
+
+impl NFT {
+    fn mint(&mut self) {
+        self.token.owner_by_id.insert(self.token_id, msg::source());
```

Comment:
we may want to limit users who have privilege for minting new tokens.

Reply:
There may be different use cases. Limiting users can be easily added, if it's required